pulp_container/app/viewsets.py

```diff
@@ -938,25 +939,31 @@ def build_image(self, request, pk):
 
         serializer.is_valid(raise_exception=True)
 
-        containerfile = serializer.validated_data["containerfile_artifact"]
-        try:
-            containerfile.save()
-        except IntegrityError:
-            containerfile = Artifact.objects.get(sha256=containerfile.sha256)
-            containerfile.touch()
+        containerfile_pk = None
+        if containerfile := serializer.validated_data.get("containerfile_artifact", None):
+            try:
+                containerfile.save()
+            except IntegrityError:
+                containerfile = Artifact.objects.get(sha256=containerfile.sha256)
+                containerfile.touch()
+            containerfile_pk = str(containerfile.pk)
+
         tag = serializer.validated_data["tag"]
+        containerfile_name = serializer.validated_data.get("containerfile_name", None)
 
-        artifacts = serializer.validated_data["artifacts"]
-        Artifact.objects.filter(pk__in=artifacts.keys()).touch()
+        build_context_pk = None
+        if build_context := serializer.validated_data.get("build_context", None):
+            build_context_pk = build_context.pk
```

Comment:
We need to put the build contexts repository (not the repository version) into the shared_resources for this task.